inc/utype.h

```diff
@@ -145,6 +147,35 @@ extern struct arabicforms {
     unsigned int required_lig_with_alef: 1;
 } ArabicForms[256];	/* for chars 0x600-0x6ff, subtract 0x600 to use array */
 
```

Comment:
What I was trying to ask was why we were merging the header files but not the code? It makes tracing function definitions slightly harder, and it seems to me that the old header was already included in all of the necessary places.

I didn't realize how much of the new file you authored. Is it divisible enough for you to mark off your part and dual-license that part?